### PR TITLE
Timeline

### PR DESCRIPTION
**Pull request by brollin**

### CODE REVIEW

**Comment by brollin** [comment]
image

**Comment by brollin** [comment]
image

**Review by jeffawang — APPROVED** [review]
wow, this looks great. It's really satisfying to watch it go!

File: modules/components/Ruler.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        <rect
+          width="0.5"
+          height="3240"
+          x={`${PIXELS_PER_SECOND * 0.75}`}
+          y="30"
+        />
+      </pattern>
+    </defs>
+    <rect
+      fill="url(#mix-editor-grid-pattern)"
```
[the remark this answers is not in the document]
Comment (brollin) [reply modules/components/Ruler.tsx:36]: I stole it from https://www.bandlab.com :D

**Pull request merged**